components/dash-core-components/tests/integration/graph/test_graph_varia.py

```diff
@@ -847,3 +859,77 @@ def graph_dims():
     assert graph_dims() == responsive_size
 
     assert dash_dcc.get_logs() == []
+
+
+def test_grva010_external_mathjax_prevents_lazy(dash_dcc):
+    app = Dash(
+        __name__,
+        eager_loading=False,
+        external_scripts=["https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-svg.js"],
+    )
+
+    app.layout = html.Div(id="div", children=[html.Button(id="btn")])
+
+    @app.callback(Output("div", "children"), [Input("btn", "n_clicks")])
+    def load_chart(n_clicks):
+        if n_clicks is None:
+            raise PreventUpdate
+
+        return dcc.Graph(mathjax=True, id="output", figure={"data": [{"y": [3, 1, 2]}]})
+
+    dash_dcc.start_server(
+        app,
+        debug=True,
+        use_reloader=False,
+        use_debugger=True,
+        dev_tools_hot_reload=False,
+    )
+
+    # Give time for the async dependency to be requested (if any)
+    time.sleep(2)
+
+    scripts = dash_dcc.driver.find_elements(By.CSS_SELECTOR, "script")
+    assert findSyncMathJax(scripts) is None
+    assert findAsyncMathJax(scripts) is None
+
+    dash_dcc.find_element("#btn").click()
+
+    # Give time for the async dependency to be requested (if any)
+    time.sleep(2)
+
+    scripts = dash_dcc.driver.find_elements(By.CSS_SELECTOR, "script")
+    assert findSyncMathJax(scripts) is None
+    assert findAsyncMathJax(scripts) is None
+
+    assert dash_dcc.get_logs() == []
+
+
+def test_grva011_without_mathjax(dash_dcc):
+    app = Dash(__name__, eager_loading=False, assets_folder="../../assets")
+
+    app.layout = html.Div([dcc.Graph(id="output", figure={"data": [{"y": [3, 1, 2]}]})])
+
+    dash_dcc.start_server(app)
+    dash_dcc.percy_snapshot("grva011 - graph without mathjax")
```

Comment:
Here too, no need for a snapshot (we test these visuals on the plotly.js side), but let's put something in the graph that would render as math and test that it shows up as plain text, as well as testing that `window.MathJax` doesn't exist.

Reply:
Also maybe we should parametrize `eager_loading` like `grva001` does, and assert `find(Sync|Async)MathJax` gives the expected results? I guess in the eager case `window.MathJax` will be defined, but the math should still not render.

We should parametrize `eager_loading` in grva012 too - and again, I'd probably drop the snapshot and just make some DOM assertions to show that MathJax renders... but if you keep it after parametrizing note you'd need to make the names unique.